Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/fragments/CameraFragment.kt

```diff
@@ -0,0 +1,488 @@
+        assert(true == characteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP)
+                ?.getHighSpeedVideoFpsRangesFor(Size(args.width, args.height))?.contains(fpsRange))
+
+        // Sends the capture request as frequently as possible until the session is torn down or
+        // session.stopRepeating() is called
+        session.setRepeatingBurst(previewRequestList, null, cameraHandler)
+
+        // Listen to the capture button
+        capture_button.setOnTouchListener { view, event ->
+            when (event.action) {
+
+                MotionEvent.ACTION_DOWN -> lifecycleScope.launch(Dispatchers.IO) {
+
+                    // Prevents screen rotation during the video recording
+                    requireActivity().requestedOrientation =
+                            ActivityInfo.SCREEN_ORIENTATION_LOCKED
+
+                    // Stops preview requests, and start record requests
+                    session.stopRepeating()
+                    session.setRepeatingBurst(recordRequestList, null, cameraHandler)
+
+                    // Finalizes recorder setup and starts recording
+                    recorder.apply {
+                        // Sets output orientation based on current sensor value at start time
+                        relativeOrientation.value?.let { setOrientationHint(it) }
+                        prepare()
+                        start()
+                    }
+                    recordingStartMillis = System.currentTimeMillis()
+                    Log.d(TAG, "Recording started")
+
+                    // Starts recording animation
+                    overlay.post(animationTask)
+                }
+
+                MotionEvent.ACTION_UP -> lifecycleScope.launch(Dispatchers.IO) {
+
+                    // Unlocks screen rotation after recording finished
+                    requireActivity().requestedOrientation =
+                            ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+
+                    // Requires recording of at least MIN_REQUIRED_RECORDING_TIME_MILLIS
+                    val elapsedTimeMillis = System.currentTimeMillis() - recordingStartMillis
+                    if (elapsedTimeMillis < MIN_REQUIRED_RECORDING_TIME_MILLIS) {
+                        delay(MIN_REQUIRED_RECORDING_TIME_MILLIS - elapsedTimeMillis)
+                    }
+
+                    Log.d(TAG, "Recording stopped. Output file: $outputFile")
+                    recorder.stop()
+
+                    // Removes recording animation
+                    overlay.removeCallbacks(animationTask)
+
+                    // Broadcasts the media file to the rest of the system
+                    MediaScannerConnection.scanFile(
+                            view.context, arrayOf(outputFile.absolutePath), null, null)
+                    Log.d("Rokus Logs:", "reached intent call in kotlin")
+                    // Launch external activity via intent to play video recorded using our provider
```

Comment:
This functionality from the boilerplate was suppressed and I added the following code to connect both the parts. i.e by using an `intent` to another activity.